### PR TITLE
🥞 Fix DPO gradient accumulation loss scaling

### PR DESCRIPTION
**Pull request by winglian**
fixes issue reported in https://github.com/huggingface/transformers/issues/35856

### CODE REVIEW

**Review by kashif — APPROVED** [review]

File: trl/trainer/dpo_trainer.py [hunk 1]
```diff
@@ -480,6 +480,9 @@ def make_inputs_require_grad(module, input, output):
             preprocess_logits_for_metrics=preprocess_logits_for_metrics,
         )
 
+        # workaround for base trainer gradient accumulation scaling
```
Comment (qgallouedec) [inline trl/trainer/dpo_trainer.py:483]:
```suggestion
        # Gradient accumulation requires scaled loss. Normally, loss scaling in the parent class depends on whether the
        # model accepts loss-related kwargs. Since we compute our own loss, this check is irrelevant. We set
        # self.model_accepts_loss_kwargs to False to enable scaling.
```

**Review by qgallouedec — CHANGES_REQUESTED** [review]

**Review by qgallouedec — APPROVED** [review]
Thanks a lot

File: trl/trainer/dpo_trainer.py [hunk 2]
```diff
@@ -480,6 +480,13 @@ def make_inputs_require_grad(module, input, output):
             preprocess_logits_for_metrics=preprocess_logits_for_metrics,
         )
 
+        # Gradient accumulation requires scaled loss. Normally, loss scaling in the parent class depends on whether the
+        # model accepts loss-related kwargs. Since we compute our own loss, this check is irrelevant. We set
+        # self.model_accepts_loss_kwargs to False to enable scaling.
+        # model accepts loss-related kwargs. Since we compute our own loss, this check is irrelevant. We set
+        # self.model_accepts_loss_kwargs to False to enable scaling.
```
Comment (qgallouedec) [inline trl/trainer/dpo_trainer.py:487]:
```suggestion
        # Gradient accumulation requires scaled loss. Normally, loss scaling in the parent class depends on whether the
        # model accepts loss-related kwargs. Since we compute our own loss, this check is irrelevant. We set
        # self.model_accepts_loss_kwargs to False to enable scaling.
```
